PWGCF/EbyEFluctuations/Tasks/nfMomentsTask.cxx

```diff
@@ -0,0 +1,389 @@
+// Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+///
+/// \brief This task is for Normalized Factorial Moments Analysis: PRC 85, 044914 (2012), nucl-ex:1411.6083
+/// \author Salman Malik
+/// \author Balwan Singh
+
+#include <iostream>
+#include <array>
+#include <TH1F.h>
+// O2 includes
+#include "Framework/AnalysisDataModel.h"
+#include "Framework/AnalysisTask.h"
+#include "Framework/runDataProcessing.h"
+#include "Common/DataModel/Multiplicity.h"
+#include "Common/DataModel/EventSelection.h"
+#include "Common/DataModel/Centrality.h"
+#include "Common/Core/TrackSelection.h"
+#include "Common/DataModel/TrackSelectionTables.h"
+
+using std::array;
+using namespace o2;
+using namespace o2::framework;
+using namespace o2::framework::expressions;
+
+namespace o2::aod
+{
+namespace full
+{
+DECLARE_SOA_COLUMN(mEvMult, eventMult, Int_t);
+DECLARE_SOA_COLUMN(mVertexZ, vertexZ, Float_t);
+DECLARE_SOA_COLUMN(mCentrality, cent, Float_t);
+DECLARE_SOA_COLUMN(mBinMult, binmult, Int_t[5]);
+DECLARE_SOA_COLUMN(mAvbCon, avbcontent, Double_t[200]);
+DECLARE_SOA_COLUMN(mValM, m, Int_t[200]);
+DECLARE_SOA_COLUMN(mPtBin, ptbin, Int_t[200]);
+DECLARE_SOA_COLUMN(mF2, f2, Double_t[200]);
+DECLARE_SOA_COLUMN(mF3, f3, Double_t[200]);
+DECLARE_SOA_COLUMN(mF4, f4, Double_t[200]);
+DECLARE_SOA_COLUMN(mF5, f5, Double_t[200]);
+DECLARE_SOA_COLUMN(mF6, f6, Double_t[200]);
+DECLARE_SOA_COLUMN(mF7, f7, Double_t[200]);
```

Comment:
You are considering using derived data as your output data. This is not the aim of the O2 framework
I'm not even sure if these large array will be supported in the near term as part of the data model 
Have in mind that if you start doing things in this way you can confront a situation where your derived data are too large to be accepted as derived data and you will at that point need to redefine all your analysis structure

Reply:
Hi @victor-gonzalez , thank you for the review. We want to store these fq(q=1,2,3,4,5,6,7) event by event for an offline analysis. We used tuples to achieve this in AliPhysics. I don't know,  if that can be implemented in O2? or do you have any suggestion? In that case, i can change it. :)

Regards,
Salman

Reply:
I don't know about the way this was done in AliPhysics but there are a few things that need to be considered in O2 and Run 3
- the table to be populated will use 14KB per collision
- for Run 3 pp collisions, there are runs which collected 2-3 Billion collisions
- for each of these runs about 20 TB will be produced
This will not be able to run on hyperloop neither as derived data producer nor as a final analysis result producer

I would suggest to plan your analysis either to minimize the post-processing (the task should produce almost final results) or to minimize the amount of information needed for doing the post-processing (but having in mind that with runs with 2-3 B collisions there is not much margin for collecting information for storing it)

Reply:
Hi @victor-gonzalez, thank you for the suggestion. I will change the analysis structure and remove the derived data part, so that we don't have issues in future. Until then, i will change the PR to WIP.